engine/src/ast/index_expr.rs

```diff
@@ -83,16 +99,37 @@ impl<'s> IndexExpr<'s> {
         }
     }
 
+    pub fn compile_with<F: 's>(
+        self,
+        default_one: bool,
+        default_vec: &'s [bool],
```

Comment:
This feels weird to me, having two arguments for each default variant.

Reply:
I prefer to have explicit default value for both cases than implicit hidden default value for the `vec` case.